src/connection.rs

```diff
@@ -207,4 +217,59 @@ impl Connection {
             },
         }
     }
+
+    fn send_query(&self, tree: &Tree) -> Result<(), Error> {
+        let token = self.query_token.get();
+        // Increment the token for the next request.
+        self.query_token.set(token.wrapping_add(1));
+        let tree = my_try!(json::encode(&tree.to_json()));
+        let len = tree.as_bytes().len();
+        if len > (u32::MAX as usize) {
+            return Err(Error::QueryTooLarge(len));
+        }
+
+        my_try!(self.stream.borrow_mut().write_u64::<LittleEndian>(token));
+        my_try!(self.stream.borrow_mut().write_u32::<LittleEndian>(len as u32));
```

Comment:
It lives in [macros.rs](https://github.com/jimberlage/rethinkdb.rs/blob/master/src/macros.rs).  There's some comment on how it works in my [documentation PR](https://github.com/jimberlage/rethinkdb.rs/pull/11).